lib/vim-plugin.nix

```diff
@@ -5,7 +5,7 @@ with lib;
     config:
     {
       name,
-      url ? if defaultPackage != null then defaultPackage.meta.homepage else null,
+      url ? if packageOption == null then null else packageOption.default.meta.homepage,
```

Comment:
nit: can't this be done with a `or` ?

Reply:
Actually this isn't needed; when I first started work on this `defaultPackage` and `packageOption` were nullable and the `package` option would be omitted from `options`.

That behaviour is never used in nixvim, so I dropped it; `packageOption` is no longer nullable.

To answer your original question, using `or` would cause `packageOption.default.meta.homepage` to silently be null when the package has no homepage, however we'd want that to error so the maintainer knows he has to set `url` explicitly.